moebot_bot/util/db/scheduledOperation.go

```diff
@@ -0,0 +1,116 @@
+package db
+
+import (
+	"log"
+	"time"
+)
+
+type SchedulerType int
+
+const (
+	SchedulerChannelRotation SchedulerType = iota
+)
+
+type ScheduledOperation struct {
+	ID                   int64
+	ServerID             int
+	Type                 SchedulerType
+	PlannedExecutionTime time.Time
+}
+
+const (
+	scheduledOperationTable = `CREATE TABLE IF NOT EXISTS scheduled_operation(
+		id SERIAL NOT NULL PRIMARY KEY,
+		server_id INTEGER NOT NULL REFERENCES server(id) ON DELETE CASCADE,
+		type INTEGER NOT NULL,
+		planned_execution_time TIMESTAMP NOT NULL DEFAULT CURRENT_TIME,
+		execution_interval INTERVAL NOT NULL
+	)`
+
+	scheduledOperationQueryNow = `SELECT id, server_id, planned_execution_time FROM scheduled_operation WHERE planned_execution_time < CURRENT_TIMESTAMP`
+
+	scheduledOperationQueryServer = `SELECT id, server_id, planned_execution_time FROM scheduled_operation WHERE server_id = $1`
+
+	scheduledOperationUpdate = `UPDATE scheduled_operation SET planned_execution_time = planned_execution_time + execution_interval WHERE id = $1`
+
+	scheduledOperationDelete = `DELETE FROM scheduled_operation WHERE id = $1 AND server_id = $2`
+
+	scheduledOperationInsert = `INSERT INTO scheduled_operation (server_id, type, execution_interval) VALUES ($1, $2, $3) RETURNING id, CURRENT_TIME + execution_interval`
+)
+
+func scheduledOperationCreateTable() {
+	moeDb.Exec(scheduledOperationTable)
+}
+
+func ScheduledOperationQueryNow() ([]*ScheduledOperation, error) {
+	rows, err := moeDb.Query(scheduledOperationQueryNow)
+	if err != nil {
+		log.Println("Error querying for current scheduled operations", err)
+		return nil, err
+	}
+	result := []*ScheduledOperation{}
```

Comment:
These should be converted into `var result []*ScheduledOperation` since we don't really do anything with the allocation.